TAScheduler/UnitTests/tests.py

```diff
@@ -10,40 +12,180 @@
 # SEE METHOD DESCRIPTIONS FOR GUIDE ON HOW TO WRITE.
 # Feel free to make suggestions on discord (add/remove/edit methods)!.
 ### Rememeber: These methods were made before any coding (I was guessing) so it's likely they should be changed.
-class TestUserLogin(TestCase):  # Alec
-    pass
-
-
-class TestUserGetID(TestCase):  # Alec
-    pass
-
-
-class TestUserGetPassword(TestCase):  # Alec
-    pass
-
-
-class TestUserGetName(TestCase):  # Alec
-    pass
-
-
-class TestUserGetRole(TestCase):  # Alec
-    pass
-
-
-class TestAdminInit(TestCase):
-    pass
-
-class TestAdminCreateCourse(TestCase):  # Alec
-    pass
-
-
-class TestAdminCreateUser(TestCase):  # Alec
-    pass
-
-
-class TestAdminCreateSection(TestCase):  # Alec
-    pass
-
+class TestUserLogin(TestCase): # Alec
+    def setUp(self):
+        user_info = User.objects.create(
+            email_address="test@example.com",
+            password="password123",
+            first_name="Test",
+            last_name="User",
+            home_address="123 Street",
+            phone_number=1234567890
+        )
+        self.userObj = UserObj(user_info)
+
+    def test_login_valid_credentials(self):
+        result = self.userObj.login("test@example.com", "password123")
+        self.assertRedirects(result, '/home/')
+
+    def test_login_invalid_credentials(self):
+        result = self.userObj.login("test@example.com", "wrongpassword")
+        self.assertRedirects(result, '/')
+class TestUserGetID(TestCase): # Alec
+    def setUp(self):
+        user_info = User.objects.create(
+            email_address="test@example.com",
+            password="password123",
+            first_name="Test",
+            last_name="User",
+            home_address="123 Street",
+            phone_number=1234567890
+        )
+        self.userObj = UserObj(user_info)
+
+    def test_get_username(self):
+        user_id = self.userObj.getUsername()
+        self.assertEqual(user_id, "test@example.com")
+class TestUserGetPassword(TestCase): # Alec
+    def setUp(self):
+        user_info = User.objects.create(
+            email_address="testuser@example.com",
+            password="password123",
+            first_name="Test",
+            last_name="User",
+            home_address="123 Street",
+            phone_number=1234567890
+        )
+        self.userObj = UserObj(user_info)
+
+    def test_get_password(self):
+        self.assertEqual(self.userObj.getPassword(), "password123")
```

Comment:
Same comment, also having a message to let us know what should have happened is nice to have, like (Passwords did not match

Reply:
can you explain a little about how I might do this? just a short snippet of what you mean?

Reply:
(mostly confused on formatting)
is it assertRedirects(... , msg="blah blah")?